### ISSUE TITLE
WCAG 2.2: Revisit button focus state

### ISSUE THREAD

**Issue by owenatgov** [alphagov/govuk-frontend#4469]
## What
Revisit the focus state of the [button component](https://design-system.service.gov.uk/components/button/).

## Why
During testing of this component against [WCAG 2.2 criterion 14.2.13: focus appearance](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html) as part of https://github.com/alphagov/govuk-frontend/issues/4347, it was discovered that the secondary and inverse button variants don't pass this criterion. This is because for both variants, the contrast between the focus yellow and the default background of these buttons (grey and white) is too low.

### Further reading
- [Testing spreadsheet (internal only)](https://docs.google.com/spreadsheets/d/1PpKvhCGZJb1j0ZUUY2ZuA0CBQeu04yFD1bumiL8wGX8/edit#gid=0)
- https://github.com/alphagov/govuk-design-system/issues/3007 - related issue on focus state rationalisation
 
## Who needs to work on this
Developers, designers

## Who needs to review this
Developers, designers

**Comment by selfthinker** [comment]
While auditing the GOV.UK Brand Guidelines website I also came across this issue. In that case it nearly failed 2.4.7 Focus Visible.

The focus state of the button on its homepage (which is the same as the 'Get started' button on the GOV.UK Design System homepage, which has the same colours as the inverse button) changes from `#1E71B8` on `#FFFFFF` to `#0B0C0C` on `#FFDD04`. That's a contrast ratio change of 3.8:1 and 1.3:1. The blue to black change is just about good enough to pass WCAG Focus Visible requirements (as it's more than 3:1).

But in the case of the secondary button the button changes from `#0B0C0C` on `#F3F2F1` to `#0B0C0C` on `#FFDD04`. That's a contrast ratio change of 1:1 and 1.2:1. Which does fail 2.4.7 Focus Visible.

Technically, WCAG doesn't define which colour contrast changes fail Focus Visible. It only says it's a fail if it's "difficult to see" (as suggested in [F78](https://www.w3.org/WAI/WCAG21/Techniques/failures/F78)). But due to other attempts to define it better (which all failed) the threshold is often 3:1 but up to an auditor's discretion.

This focus style also doesn't follow our usual focus state guidance to have a thicker bottom border. That thicker border would improve accessibility of the focus state. Although probably not if it extends to the bottom when the background is already dark (like black on blue in the case of the inverse button) and there is no dark to light background change on the button.

**Comment by selfthinker** [comment]
> But in the case of the secondary button the button changes from `#0B0C0C` on `#F3F2F1` to `#0B0C0C` on `#FFDD04`. That's a contrast ratio change of 1:1 and 1.2:1. Which does fail 2.4.7 Focus Visible.

I've discussed this internally with other accessibility specialists and we've agreed that this does actually **not** fail 2.4.7 Focus Visible. That is because the colour contrast changes on the border from middle grey (`#929191`) to black (`#0B0C0C`) border which is a contrast of 6.2:1. We think that is enough to not fail WCAG.

Although this is clearly still an accessibility issue as the change is not very visible, certainly not as visible as our usual focus changes.

I will adjust the description a bit to add more information and merge the issues for the two different button styles.

**Comment by selfthinker** [comment]
I have removed the link to the [testing spreadsheet (internal only)](https://docs.google.com/spreadsheets/d/1PpKvhCGZJb1j0ZUUY2ZuA0CBQeu04yFD1bumiL8wGX8/edit#gid=0) from the description because it is outdated and not relevant anymore. But I'm keeping a link in this comment for historical reasons.

**Comment by mia-allers-gds** [comment]
@Kam-Nijjar I am putting this back into the backlog because while I have been working on it, I think I still don't quite understand the problem. For mid-cycle check in, please make this issue AMBER and add in a note that I will look at it in week 3 but I can't guarantee anything will be shipped.

**Comment by selfthinker** [comment]
Although it would be good to meet Focus Appearance with these buttons, it's worth pointing out that the focus state of our links don't pass Focus Appearance either. Therefore I think it's fair if we focus on improving the visibility of the secondary and inverse button focus states without necessarily passing that criterion.

**Comment by selfthinker** [comment]
I think for a fix for this we should not focus on WCAG but rather how we can generally improve the distinction between the normal and the focus states of these buttons. One method is to look at both states in greyscale and make them distinct enough that way.

For example, this is what the two states of the secondary button looks like next to each other using a greyscale filter:

<img width="274" height="61" alt="Secondary button normal state next to focus state in greyscale" src="https://github.com/user-attachments/assets/969c7699-33da-49f3-84fc-c8126b8fbf53" />
